Lib/test/test_traceback.py

```diff
@@ -669,16 +705,14 @@ class CPythonTracebackErrorCaretTests(TracebackErrorLocationCaretTests):
     Same set of tests as above but with Python's internal traceback printing.
     """
     def get_exception(self, callable):
-        from _testcapi import traceback_print
+        from _testcapi import exception_print
         try:
             callable()
             self.fail("No exception thrown.")
-        except:
-            type_, value, tb = sys.exc_info()
-
-            file_ = StringIO()
-            traceback_print(tb, file_)
-            return file_.getvalue().splitlines()
+        except Exception as e:
+            with captured_output("stderr") as tbstderr:
+                exception_print(e)
+            return tbstderr.getvalue().splitlines()[:-1]
```

Comment:
I imagine you add the[:-1] to remove the exception type+msg line, but this doesn't make the function equivalent to what it was before - exception_print would print also chained exception, and the exception msg can be more than one line, and once PEP 678 is accepted there will also be notes. 

So I think if we're going for this (which seems ok for now because all the tests are passing), then maybe without the [:-1]?

Reply:
The [:-1] keeps the output of the CPythonTracebackErrorCaretTests and TracebackErrorLocationCaretTests implementations of get_exception() the same. This is the TracebackErrorLocationCaretTests implementation:

```
class TracebackErrorLocationCaretTests(unittest.TestCase):
    """
    Tests for printing code error expressions as part of PEP 657
    """
    def get_exception(self, callable):
        try:
            callable()
            self.fail("No exception thrown.")
        except:
            return traceback.format_exc().splitlines()[:-1]
```

If we remove the [:-1] from both implementations we immediately have 26 test failures.

It is certainly possible to amend the test expectations to cover the exception messages being returned. I'm not sure why the tests did not do this from the start.

My question is if you think this should be done as part of this issue?

Reply:
I see now that the python version of get_exception TracebackErrorLocationCaretTests does the same thing. I think it's fine.